logstash-core/src/main/java/org/logstash/common/io/wip/MemoryPageIOStream.java

```diff
@@ -70,6 +71,11 @@ public MemoryPageIOStream(int capacity, byte[] initialBytes) throws IOException
     }
 
     @Override
+    public void recover() throws IOException {
+        throw new NotImplementedException();
```

Comment:
Curiosity - Is this the right exception to throw, or should we make our own class for this?